From 2b71016a3c3d4c088e8edd170fe6eb8431fd71fa Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Mon, 4 Jun 2012 12:52:14 +0200
Subject: [PATCH] fstab-generator: avoid mangling of non-path mount source and
 dest

This can invalidate otherwise valid source paths with trailing slashes,
such as "host:/" in the case of a network mount.

Based on a patch from Dave Reisner <dreisner@archlinux.org>, which
removed the slash mangling entirely.

Changed it to match on the leading slash to exclude non-path values.
---
 src/fstab-generator/fstab-generator.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fstab-generator/fstab-generator.c b/src/fstab-generator/fstab-generator.c
index 8419a0c..0746724 100644
--- a/src/fstab-generator/fstab-generator.c
+++ b/src/fstab-generator/fstab-generator.c
@@ -470,10 +470,10 @@ static int parse_fstab(void) {
                         goto finish;
                 }
 
-                if (is_path(what))
+                if (path_is_absolute(what))
                         path_kill_slashes(what);
 
-                if (is_path(where))
+                if (path_is_absolute(where))
                         path_kill_slashes(where);
 
                 log_debug("Found entry what=%s where=%s type=%s", what, where, me->mnt_type);
-- 
1.7.10.3

From ec6ceb18663940efb1963704923430be0e83f1f7 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Mon, 4 Jun 2012 15:21:05 +0200
Subject: [PATCH] fstab-generator: avoid mangling of mount source string

This is a valid source entry in /etc/fstab:
  //192.168.6.10/data /data cifs noperm,auto

On Mon, Jun 4, 2012 at 2:04 PM, Dave Reisner <d@falconindy.com> wrote:
> On Mon, Jun 04, 2012 at 12:57:47PM +0200, Kay Sievers wrote:
>>
>> Changed it to use:
>>   path_is_absolute()
>> instead of:
>>   is_path(),
>> so that we still sanitize the input we might match against.
>>
>> Let me know, if you think that could still cause any problems?

> Yes, this will still break CIFS shares.
---
 src/fstab-generator/fstab-generator.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/fstab-generator/fstab-generator.c b/src/fstab-generator/fstab-generator.c
index 0746724..d9ba3e3 100644
--- a/src/fstab-generator/fstab-generator.c
+++ b/src/fstab-generator/fstab-generator.c
@@ -470,10 +470,7 @@ static int parse_fstab(void) {
                         goto finish;
                 }
 
-                if (path_is_absolute(what))
-                        path_kill_slashes(what);
-
-                if (path_is_absolute(where))
+                if (is_path(where))
                         path_kill_slashes(where);
 
                 log_debug("Found entry what=%s where=%s type=%s", what, where, me->mnt_type);
-- 
1.7.10.3

